src/frontend/deployment_details.go

```diff
@@ -3,27 +3,37 @@ package main
 import (
 	"net/http"
 	"os"
-	"sync"
+	"time"
 
 	"cloud.google.com/go/compute/metadata"
 	"github.com/sirupsen/logrus"
 )
 
-const (
-	DetailsAreNotLoaded = 0
-	DetailsAreLoading   = 1
-	DetailsAreLoaded    = 2
-)
-
+var areDeploymentDetailsLoaded = false
 var deploymentDetailsMap = make(map[string]string)
-var detailsLoadingStatus = DetailsAreNotLoaded
-var detailsLoadingStatusMutex sync.Mutex // Ensures only 1 HTTP request loads deployment details.
+var log *logrus.Logger
 
-func loadDeploymentDetails(httpRequest *http.Request) map[string]string {
-	var metaServerClient = metadata.NewClient(&http.Client{})
+func init() {
+	initializeLogger()
+	go loadDeploymentDetails()
+}
 
-	// The use of httpRequest here lets us to see HTTP request info in the logs.
-	var log = httpRequest.Context().Value(ctxKeyLog{}).(logrus.FieldLogger)
+func initializeLogger() {
```

Comment:
I'm okay with having a single logger, but if so:
- Let's put it in either `main.go` or its own `logger.go`, and
- Let's use it everywhere. I notice some files (like `main.go` uses their own duplicated loggers)

Reply:
I was thinking the same thing.
Let's track this in a different issue: https://github.com/GoogleCloudPlatform/microservices-demo/issues/703
I'd say it's outside the scope of this P1.